streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -359,11 +359,8 @@ public static StreamThread create(final InternalTopologyBuilder builder,
         consumerConfigs.put(StreamsConfig.InternalConfig.ASSIGNMENT_ERROR_CODE, assignmentErrorCode);
         final AtomicLong nextScheduledRebalanceMs = new AtomicLong(Long.MAX_VALUE);
         consumerConfigs.put(StreamsConfig.InternalConfig.NEXT_SCHEDULED_REBALANCE_MS, nextScheduledRebalanceMs);
-        String originalReset = null;
-        if (!builder.latestResetTopicsPattern().pattern().equals("") || !builder.earliestResetTopicsPattern().pattern().equals("")) {
-            originalReset = (String) consumerConfigs.get(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG);
-            consumerConfigs.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
-        }
```

Comment:
I had to change this to get the StreamThread test to actually use the configured `ConsumerConfig.AUTO_OFFSET_RESET_CONFIG := earliest`

Reading the conditional, it doesn't make any sense to me, but it's been in the codebase for a long time, so I'm doubting myself. It seems to say that we will only use the provided client configuration if there **is** an override, but it seems like it should have been "if there is **not** an override".

Regardless, the "originalReset" is only used as a fallback _after_ we apply the builder reset patterns, so I don't see why we should leave it null in any case.

Reply:
This makes my head hurt..

Reply:
If users specify an in-code overwrite via `Consumed` the user might do this only for _some_ topics. For all other topics, the configures reset policy should be used. Hence, we just "remember" the default config in `originalReset` and set the consumer config to "none".

If there is no in-code overwrite, we don't need to remember the original strategy because all topics use the same strategy and thus we rely on the consumer anyway to do the reset for us.

With this change, we now _always_ set the reset policy to "none". What is still possible of course, but it implies, we never rely on the consumer any longer to do any reset and we always to it manually. (This might be "cleaner" as it might be easier to reason about.) -- But the old code was correct, too.

Reply:
I guess the main reason for this old design was, that user see their config when `ConsumerConfig` is logged. -- Now, we would always log "none" independent what the users sets, and this might be confusing.

I think we should update the docs accordingly and call it out.

Reply:
Aha! Thanks, @mjsax. This makes sense. I hadn't considered that the consumer never throws `InvalidOffsetException` if there's a valid policy configured, so the prior logic only needs the fallback when there's an override.

Now that we also use the resetting logic for our own "manual" reset when handling TaskCorruptedException, we do need to capture the Consumer config value, but we can still enforce that if there are any overrides in the topology, we set the consumer to "none". Just updated the code to do this.